proto/regen/ecocredit/v1alpha1/tx.proto

```diff
@@ -367,54 +344,8 @@ message MsgBuy {
   string buyer = 1;
 
   // orders are the new buy orders.
-  repeated Order orders = 2;
-
-  // Order is a buy order.
-  message Order {
-
-    // Selection defines a buy order selection.
-    message Selection {
-
-      // sum defines the type of selection.
-      oneof sum {
-        // sell_order_id is the sell order ID against which the buyer is trying to buy.
-        // When sell_order_id is set, this is known as a direct buy order because it
-        // is placed directly against a specific sell order.
-        uint64 sell_order_id = 1;
-
-        // TODO: once we have filters defined for baskets, we can enable filtered
-        // buy orders which will involve full double-sided batch order matching
-        //
-        // filter selects credits to buy based upon the specified filter criteria.
-        // Filter filter = 2;
-      }
-    }
-
-    // selection is the buy order selection.
-    Selection selection = 1;
-
-    // quantity is the quantity of credits to buy. If the quantity of credits
-    // available is less than this amount the order will be partially filled
-    // unless disable_partial_fill is true.
-    string quantity = 2;
-
-    // bid price is the bid price for this buy order. A credit unit will be
-    // settled at a purchase price that is no more than the bid price. The
-    // buy order will fail if the buyer does not have enough funds available
-    // to complete the purchase.
-    cosmos.base.v1beta1.Coin bid_price = 3;
-
-    // disable_auto_retire allows auto-retirement to be disabled. If it is set to true
-    // the credits will not auto-retire and can be resold assuming that the
-    // corresponding sell order has auto-retirement disabled. If the sell order
-    // hasn't disabled auto-retirement and the buy order tries to disable it,
-    // that buy order will fail.
-    bool disable_auto_retire = 4;
+  repeated BuyOrder orders = 2;
```

Comment:
I think this works as is but as mentioned above, I think we actually need to add an `owner` or `buyer` field to `BuyOrder` and then we should support the ability to query buy orders by address. This should have been done in #613 but maybe we can add that here. Thoughts?

Reply:
yeah i think it makes sense to embed address into buy order. I'll add it in this PR